htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -70,206 +167,139 @@ def killCells(self, percent = 0.05):
+      if numActivePotential[i] >= self.minThreshold and
+         not self.connections.cells[self.connections.segmentForFlatIdx(i).cell].destroyed
+    )
+
+    maxSegmentsPerCell = self.connections.maxSegmentsPerCell
+    segmentKey = lambda segment: (segment.cell * maxSegmentsPerCell
+                                  + segment.idx)
+    self.activeSegments = sorted(activeSegments, key=segmentKey)
+    self.matchingSegments = sorted(matchingSegments, key=segmentKey)
+    self.numActiveConnectedSynapsesForSegment = numActiveConnected
+    self.numActivePotentialSynapsesForSegment = numActivePotential
+
+    if learn:
+      for segment in self.activeSegments:
+        self.connections.recordSegmentActivity(segment)
+      self.connections.startNewIteration()
+
+
+  def burstColumn(self, connections, random, column, columnMatchingSegments,
+                  prevActiveCells, prevWinnerCells,
+                  numActivePotentialSynapsesForSegment, cellsPerColumn,
+                  maxNewSynapseCount, initialPermanence, permanenceIncrement,
+                  permanenceDecrement, learn):
+    """ Originally copied from temporal_memory.py and augmented to ignore dead
+    cells.  See TemporalMemory.burstColumns() for original implementation.
     """
-    numActiveConnectedSynapsesForSegment = defaultdict(lambda: 0)
-    numActiveSynapsesForSegment = defaultdict(lambda: 0)
-    activeSegments = set()
-    predictiveCells = set()
-
-    matchingSegments = set()
-    matchingCells = set()
-
-    for cell in activeCells:
-      assert(not (cell in self.deadCells))
-      for synapseData in connections.synapsesForPresynapticCell(cell).values():
-        segment = synapseData.segment
-        permanence = synapseData.permanence
-        postSynapticCell = connections.cellForSegment(segment)
+    start = cellsPerColumn * column
+    cells = xrange(start, start + cellsPerColumn)
 
-        if not(postSynapticCell in self.deadCells):
-          if permanence >= self.connectedPermanence:
-            numActiveConnectedSynapsesForSegment[segment] += 1
+    # Strip out destroyed cells
+    cells = [cellIdx
+             for cellIdx
+             in cells
+             if not self.connections.cells[cellIdx].destroyed]
 
-            if (numActiveConnectedSynapsesForSegment[segment] >=
-                self.activationThreshold):
-              activeSegments.add(segment)
-              predictiveCells.add(connections.cellForSegment(segment))
+    if columnMatchingSegments is not None:
+      numActive = lambda s: numActivePotentialSynapsesForSegment[s.flatIdx]
 
-          if permanence > 0 and self.predictedSegmentDecrement > 0:
-            numActiveSynapsesForSegment[segment] += 1
+      bestMatchingSegment = None
+      for seg in columnMatchingSegments:
+        if bestMatchingSegment is None:
+          bestMatchingSegment = seg
+        elif seg > bestMatchingSegment and not connections.cells[seg.cell].destroyed:
+          bestMatchingSegment = seg
```

Comment:
I'd added this because at the time, it was.  Some subsequent changes may have fixed that.  I will check again.